### PR TITLE
Weft Proposal

### PR DESCRIPTION
**Pull request by mbwhite**
Signed-off-by: Matthew B White <whitemat@uk.ibm.com>

## Checklist for Hyperledger Lab Proposal PR

- [x] Read [README.md](https://github.com/hyperledger-labs/hyperledger-labs.github.io/blob/master/README.md) carefully
- [x] Make sure your changes are committed with the proper sign-off
- [x] Use the labs name as the PR title
- [x] Copy the short description of the proposal as the description of the PR
- [x] Remove this checklist section
- [x] Submit your PR and the labs stewards will review your proposal
- [x] Ask your sponsor to confirm sponsorship by stating so in a comment to the PR

### CODE REVIEW

**Comment by jt-nti** [comment]
I'm happy to sponsor this proposal

**Review by anandbanik — COMMENTED** [review]
Hi @mbwhite,
Looks good to me, but since not all of the commits on the referenced repo are signed-off, you will have to push the code to the new repo under hyperledger-labs org with single sign-off. 
Thanks!

**Comment by anandbanik** [comment]
Also, please ensure to add the Apache 2.0 license file as mentioned [here](https://github.com/hyperledger-labs/hyperledger-labs.github.io/blob/main/LICENSE)

**Comment by jt-nti** [comment]
@anandbanik it would be a pity to lose the commit history- would it be ok to update all the commits with a sign-off instead?

**Comment by anandbanik** [comment]
Hi @jt-nti Sure. Please let me know once all the git commits are signed-off. Thanks!

File: labs/fabric-weft.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
+---
+layout: default
+title: weft
+parent: Labs
+---
+# Lab Name
+[_Fabric Weft._](https://github.com/hyperledgendary/weft)
+
+# Short Description
+Simple command line utility and module, to help work Hyperledger Fabric, and other instations such as the IBM Blockchain Platform (IBP) and convert resources between them. This is intended to be a complement to the existing tools for each product. A lot of the code here is based on that in hyperledger/fabric-samples
```
Comment (tkuhrt) [inline labs/fabric-weft.md:10]: Not sure if this is what you meant
```suggestion
Simple command line utility and module, to help work Hyperledger Fabric, and other instantiations such as the IBM Blockchain Platform (IBP) and convert resources between them. This is intended to be a complement to the existing tools for each product. A lot of the code here is based on that in hyperledger/fabric-samples
```

File: labs/fabric-weft.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+# Scope of Lab
+The end goals are to obtain, irrespective of how the Fabric infrastructure is created.
+
+- application wallets and idenities for the Client SDKs to use
```
Comment (tkuhrt) [inline labs/fabric-weft.md:15]:
```suggestion
- application wallets and identities for the Client SDKs to use
```

File: labs/fabric-weft.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+- the gateway connection profiles to let the Client SDKs to connect
+- the MSP directory structure to permit the Fabric Peer CLIs to function
+- packaged Chaincode/SmartContract
+
+
+For example packaing chaincode the cli is
```
Comment (tkuhrt) [inline labs/fabric-weft.md:21]:
```suggestion
For example packaging chaincode the cli is 
```

**Review by tkuhrt — CHANGES_REQUESTED** [review]
Before we can transfer the existing repository, we will need the following done in that repository:
1. Add the Apache LICENSE file
2. Ensure that all commits have DCO sign-off completed.

**Comment by tkuhrt** [comment]
> > Before we can transfer the existing repository, we will need the following done in that repository:
> > 
> > 1. Add the Apache LICENSE file
> > 2. Ensure that all commits have DCO sign-off completed.
> 
> @tkuhrt thanks (sorry I missed the notification) - can happily do that... personally I don't mind push everything in one go... but no harm in trying to keep the history

No worries. Whatever you want regarding DCO sign-off. Just let us know which direction you want to proceed.

**Review by tkuhrt — APPROVED** [review]

**Comment by mbwhite** [comment]
@tkuhrt there's not a lot of value in the commit history I think in this case; happy to push it all together. 

one other question - this has previously published a module to npm (under the original orgs namespace) - what's the process in hyperledger-labs?

**Comment by tkuhrt** [comment]
> @tkuhrt there's not a lot of value in the commit history I think in this case; happy to push it all together.
Sounds good.

> one other question - this has previously published a module to npm (under the original orgs namespace) - what's the process in hyperledger-labs?
@ryjones : Can you help with this question?

**Comment by anandbanik** [comment]
> > @tkuhrt there's not a lot of value in the commit history I think in this case; happy to push it all together.
> 
> Sounds good.
> 
> > one other question - this has previously published a module to npm (under the original orgs namespace) - what's the process in hyperledger-labs?
> 
> @ryjones : Can you help with this question?

@mbwhite One option is to use github actions to publish the npm module as github package under the hyperledger-labs org.
would like to hear @ryjones opinion on it.

**Comment by ryjones** [comment]
@mbwhite please transfer the repo to me and I will put it here

**Pull request merged**